diff -r 233e1a91ad72 spwinput_win32.h.in
--- a/spwinput_win32.h.in	Fri Nov 03 11:50:59 2017 +0100
+++ b/spwinput_win32.h.in	Fri Mar 09 13:58:48 2018 +0100
@@ -96,6 +96,7 @@
 
 /* the accepted datatypes for spacetec products */
 
+#if 0 // unused type definitions
 #define tchar_t				_TCHAR
 #define char_t				char
 #define uint32_t			unsigned long
@@ -103,6 +104,7 @@
 #define boolean_t			unsigned char  
 #define void_t				void
 #define window_handle_t		HWND
+#endif
 
 
 typedef long               SPWint32;
